docs/en/contributing-to-appium/developers-overview.md

```diff
@@ -0,0 +1,150 @@
+
+### Linting and Style
+
+It's important for all of Appium's JS to look and feel the same. This includes
+style conventions as well as coding patterns and which libraries we use to
+solve various problems. You should get familiar with our new [ES2015 Style
+Guide](/docs/en/contributing-to-appium/style-guide-2.0.md). When transpiling,
+Appium packages will automatically run JSHint or other lint tools and provide
+warning or error feedback if the code doesn't conform to our style. These tools
+are not necessarily exhaustive of the kinds of style issues we care about, so
+we may also mention style issues in our reviews. This isn't to be nit-picky but
+to have a clean, consistent, and readable codebase!
+
+### Submitting Code
+
+Getting your code into Appium is easy: just submit a PR to one of our repos and
+engage with the maintainers in the review process. We have a number of
+requirements for code submissions:
+
+* Follow the style of the surrounding code and our Style Guide
+* Atomic commits--one commit per logical change (i.e., make sure that commits don't need to come in a group in order for the program to work. It should work at any given commit). Usually this means one commit per PR. You'll want to get very familiar with `git rebase -i` and squashing!
+* No merge commits: always rebase on top of latest master (or whatever other branch you're asking to merge into) before submitting your PR.
+* Almost all changes should have tests. Bugfixes should at the least have unit tests that prove that the bug has been fixed. New features should have unit tests and in most cases e2e tests to prove that the feature actually works. We will be happy to walk you through the test creation process. Reading the surrounding test code is a good place to start. Our CI systems usually run test coverage statistics and we will likely not merge code that decreases test coverage.
+
+If you do all of these things before submission, your code will almost
+certainly be accepted very quickly! Of course, if you're thinking of making
+a change to Appium that requires a lot of work, you might reach out to the
+developers list to make sure that the change is in line with our philosophy and
+in principle something that we'll accept before you get going.
+
+### Testing
+
+Always make sure that your changes are tested! In addition to writing unit and
+e2e tests, make sure you run existing tests before you begin to make changes
+and before you push code to be reviewed. We do have CI set up for every Appium
+repository as a safety net for reviewers to know whether the code they are
+reviewing has passed muster. Running tests in any Appium package is easy!
+Unless the README says otherwise, here are the things you can do:
+
+```
+gulp                    # watch code for transpilation and run unit tests
+gulp once               # same as above but don't watch
+gulp unit-test          # transpile and run unit tests
+gulp e2e-test           # transpile and run e2e tests
```

Comment:
how about we say "end-to-end tests" instead of "e2e". People reading this might not be familiar with the abbreviation, but will learn it when they see the command.